pam/integration-tests/helpers_test.go

```diff
@@ -33,12 +34,15 @@ func prepareFileLogging(t *testing.T, fileName string) string {
 func requirePreviousBrokerForUser(t *testing.T, socketPath string, brokerName string, user string) {
 	t.Helper()
 
-	conn, err := grpc.Dial(
-		"unix://"+socketPath,
-		grpc.WithTransportCredentials(insecure.NewCredentials()),
-		grpc.WithBlock(),
-	)
+	conn, err := grpc.NewClient("unix://"+socketPath, grpc.WithTransportCredentials(insecure.NewCredentials()))
 	require.NoError(t, err, "Can't connect to authd socket")
+
+	conn.Connect()
+	if conn.GetState() != connectivity.Ready {
```

Comment:
I know this is mimicking the previous behaviour `WithBlock()`, but do we really need to wait for it? Does the tests still pass without this?

Reply:
We do, otherwise, we might have races. The issue is that NewClient does nothing until a request comes along and Connect does not wait for the connection to be ready, so we need to do this kind of checking :smiling_face_with_tear:

Reply:
do we really? The whole idea of gRPC is that the connection can be dropped at anytime and it will reconnect on the next call.

So, there is no real "race", it should block and connect only when needed. (this is the reason why they removed this call if I look at the reasoning)

Reply:
Locally the tests passed, let's try removing it and see if the behavior is the same in the CI.